src/client/Microsoft.Identity.Client/Platforms/netcore/NetCoreDeviceAuthManager.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http.Headers;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.Identity.Client.PlatformsCommon.Interfaces;
+using Microsoft.Identity.Client.PlatformsCommon.Shared;
+using Microsoft.Identity.Client.OAuth2;
+
+namespace Microsoft.Identity.Client.Platforms.netcore
+{
+    internal class NetCoreDeviceAuthManager : IDeviceAuthManager
+    {
+        public bool TryCreateDeviceAuthChallengeResponseAsync(HttpResponseHeaders responseHeaders, Uri endpointUri, out string responseHeader)
+        {
+            responseHeader = string.Empty;
+            X509Certificate2 certificate = null;
+
+            if (!DeviceAuthHelper.IsDeviceAuthChallenge(responseHeaders))
+            {
+                return false;
+            }
+            if (!DeviceAuthHelper.CanOSPerformPKeyAuth())
+            {
+                responseHeader = DeviceAuthHelper.GetBypassChallengeResponse(responseHeaders);
+                return true;
+            }
+
+            IDictionary<string, string> challengeData = DeviceAuthHelper.ParseChallengeData(responseHeaders);
+
+            if (!challengeData.ContainsKey("SubmitUrl"))
+            {
+                challengeData["SubmitUrl"] = endpointUri.AbsoluteUri;
+            }
+
+            try
```

Comment:
lines 18 - 49 in this file is identical to what we have in the other device auth managers. Wouldnt hurt to move these to functions in the device auth helper.
For example, lines 38 to 49 can be "GetDeviceAuthCert()" and moved to the deviceAuthHelper. or perhaps you could use some kind of inheretence. Not too big of a deal but would further remove code duplication

Reply:
Done.